### PR TITLE
don't depend on sui-types/sui-keys

### PR DESCRIPTION
**Pull request by joske**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -6,12 +6,12 @@ members = [
     "crates/mysten-util-mem",
     "crates/mysten-util-mem-derive",
     "crates/prometheus-closure-metric",
-    "crates/sui-cost-tables",
-    "crates/sui-keys",
+    # "crates/sui-cost-tables",
+    # "crates/sui-keys",
```
Comment (ljedrz) [inline Cargo.toml:10]:
```suggestion
```

File: crates/mysten-network/Cargo.toml [hunk 2]
```diff
@@ -15,7 +15,7 @@ futures = "0.3.21"
 http = "0.2.8"
 multiaddr = "0.17.0"
 serde = { version = "1.0.140", features = ["derive"] }
-tokio = { workspace = true, features = ["sync", "rt", "macros"] }
+tokio = { version = "1", features = ["full"] }
```
Comment (ljedrz) [inline crates/mysten-network/Cargo.toml:18]: I'd prefer to stick to just the required features here, I think they are fine-tuned in snarkOS as well

Comment (ljedrz) [reply crates/mysten-network/Cargo.toml:18]: this applies to other instances of the `tokio` dep as well

File: crates/typed-store/Cargo.toml [hunk 3]
```diff
@@ -22,7 +22,7 @@ hdrhistogram = "7.5.1"
 rocksdb = { version = "0.19.0", features = ["snappy", "lz4", "zstd", "zlib", "multi-threaded-cf"], default-features = false }
 serde = { version = "1.0.140", features = ["derive"] }
 thiserror = "1.0.37"
-tokio = { workspace = true, features = ["full", "test-util"] }
+tokio = { version = "1", features = ["full", "test-util"] }
```
Comment (ljedrz) [inline crates/typed-store/Cargo.toml:25]: `full` includes the components of `test-util`

File: crates/mysten-network/Cargo.toml (hunk 2 above)
Comment (joske) [reply crates/mysten-network/Cargo.toml:18]: yeah, that was a blunt axe 😅

File: crates/typed-store/Cargo.toml (hunk 3 above)
Comment (joske) [reply crates/typed-store/Cargo.toml:25]: no it doesn't. I had to add that manually to make it compile.

Comment (ljedrz) [reply crates/typed-store/Cargo.toml:25]: I mean: https://github.com/tokio-rs/tokio/blob/master/tokio/Cargo.toml#L88

Comment (niklaslong) [reply crates/typed-store/Cargo.toml:25]: I don't think so? https://github.com/tokio-rs/tokio/blob/master/tokio/Cargo.toml#L29

Comment (joske) [reply crates/typed-store/Cargo.toml:25]: Not following what you mean?

Comment (ljedrz) [reply crates/typed-store/Cargo.toml:25]: `full` contains `rt, sync, time`, which are the only contents of `test-util`

Comment (ljedrz) [reply crates/typed-store/Cargo.toml:25]: unless `test-util` is required directly, that is :detective:

Comment (niklaslong) [reply crates/typed-store/Cargo.toml:25]: I think @ljedrz was saying the `test-util` feature is included in the `full` features but it doesn't look like it is which would explain why you needed both there.

Comment (joske) [reply crates/typed-store/Cargo.toml:25]: It is needed for some macros that test code uses.

Comment (joske) [reply crates/typed-store/Cargo.toml:25]: Because it depends on conditionally compiled code depending on that feature.

Comment (ljedrz) [reply crates/typed-store/Cargo.toml:25]: yeah, makes sense; I just never saw it used explicitly in the tokio APIs and thought it was just a bundle of features used for test purposes :+1:

**Review by niklaslong — APPROVED** [review]

**Pull request merged**